timemachine/cpp/src/neighborlist.cu

```diff
@@ -76,22 +76,28 @@ void Neighborlist<RealType>::compute_block_bounds_host(
 
     assert(N == N_);
     assert(D == 3);
+    int h_rebuild[1] = {1};
 
     double *d_coords = gpuErrchkCudaMallocAndCopy(h_coords, N*3*sizeof(double));
-    double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3*3*sizeof(double));  
-   
+    double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3*3*sizeof(double));
+    int *d_rebuild = gpuErrchkCudaMallocAndCopy(h_rebuild, 1*sizeof(int));
+    cudaStream_t stream = static_cast<cudaStream_t>(0);
     this->compute_block_bounds_device(
         N,
         D,
         d_coords,
         d_box,
-        static_cast<cudaStream_t>(0)
+        d_rebuild,
+        stream
     );
-    // Does this need to peek at the last error?
-    cudaDeviceSynchronize();
 
-    gpuErrchk(cudaMemcpy(h_bb_ctrs, d_block_bounds_ctr_, this->B()*3*sizeof(*d_block_bounds_ctr_), cudaMemcpyDeviceToHost));
-    gpuErrchk(cudaMemcpy(h_bb_exts, d_block_bounds_ext_, this->B()*3*sizeof(*d_block_bounds_ext_), cudaMemcpyDeviceToHost));
+    gpuErrchk(cudaMemcpyAsync(h_bb_ctrs, d_block_bounds_ctr_, this->B()*3*sizeof(*d_block_bounds_ctr_), cudaMemcpyDeviceToHost, stream));
+    gpuErrchk(cudaMemcpyAsync(h_bb_exts, d_block_bounds_ext_, this->B()*3*sizeof(*d_block_bounds_ext_), cudaMemcpyDeviceToHost, stream));
+
+    gpuErrchk(cudaFreeAsync(d_coords, stream));
```

Comment:
I don't think the cudaMalloc's for these buffers are async

Reply:
this host code isn't really important for perf, i'd rather we use the vanilla versions of these for simplicity.

Reply:
You are right, was just trying to avoid the device sync in case someone decided to call this. Was adding the calls to free and my OCD had me change it